### DISCUSSION TITLE
can vsg::DescriptorImage generate mipmap automatically？

### DISCUSSION THREAD

**Discussion by orchidshelley** [vsg-dev/VulkanSceneGraph#847] [Q&A] [answered]
If  I used 
`auto tex = vsg::DescriptorImage::create(textureData, 0, 0, VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER);`

Does it can  generate mipmap automatically？

If it doesn't, What should I do? Can you give me some code as an example?

thank you.

**Comment by vsg-dev** [comment, marked as answer]
The VSG will generated mipmaps when required for you based if the vsg::Sampler setting indicates it's required and there no mipmaps provided by the vsg::Data.  The will be done as part of the work the VSG does to transfer data from main memory to the GPU memory.
